src/main/java/build/buildfarm/server/BuildFarmServer.java

```
@@ -116,6 +120,17 @@ private static BuildFarmServerConfig toBuildFarmServerConfig(
     return builder.build();
   }
 
+  private static MetricsPublisher getMetricsPublisher(MetricsConfig metricsConfig) {
+    switch (metricsConfig.getMetricsDestination()) {
+      default:
+        return new LogMetricsPublisher(metricsConfig);
+      case "aws":
+        return new AwsMetricsPublisher(metricsConfig);
+      case "gcp":
+        throw new UnsupportedOperationException("Not Implemented.");
```

Let the implementation throw on bad usage, you've provided the class already